flink-libraries/flink-table/src/test/scala/org/apache/flink/api/scala/stream/table/GroupWindowITCase.scala

```diff
@@ -0,0 +1,777 @@
+
+    val expected = Seq("Hello world,2", "Hello,2")
+    assertEquals(expected.sorted, StreamITCase.testResults.sorted)
+  }
+
+  @Test
+  def testProcessingTimeSlidingGroupWindowOverTime(): Unit = {
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)
+    val tEnv = TableEnvironment.getTableEnvironment(env)
+    StreamITCase.testResults = mutable.MutableList()
+
+    val data = new mutable.MutableList[(Long, Int, String)]
+    data.+=((1L, 1, "Hi"))
+    data.+=((2L, 2, "Hello"))
+    data.+=((4L, 2, "Hello"))
+    data.+=((8L, 3, "Hello world"))
+    data.+=((16L, 3, "Hello world"))
+    val stream = env.fromCollection(data)
+
+    val table = stream.toTable(tEnv, 'long, 'int, 'string)
+
+    val windowedTable = table
+      .groupBy('string)
+      .window(Slide over 50.milli every 50.milli)
+      .select('string, 'int.count)
+
+    // we only test if validation is successful here since processing time is non-deterministic
+    windowedTable.toDataStream[Row]
+  }
+
+  @Test
+  def testProcessingTimeSlidingGroupWindowOverCount(): Unit = {
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)
+    val tEnv = TableEnvironment.getTableEnvironment(env)
+    StreamITCase.testResults = mutable.MutableList()
+
+    val data = new mutable.MutableList[(Long, Int, String)]
+    data.+=((1L, 1, "Hi"))
+    data.+=((2L, 2, "Hello"))
+    data.+=((4L, 2, "Hello"))
+    data.+=((8L, 3, "Hello world"))
+    data.+=((16L, 3, "Hello world"))
+    val stream = env.fromCollection(data)
+
+    val table = stream.toTable(tEnv, 'long, 'int, 'string)
+
+    val windowedTable = table
+      .groupBy('string)
+      .window(Slide over 2.rows every 1.rows)
+      .select('string, 'int.count)
+
+    val results = windowedTable.toDataStream[Row]
+    results.addSink(new StreamITCase.StringSink)
+    env.execute()
+
+    val expected = Seq("Hello world,1", "Hello world,2", "Hello,1", "Hello,2", "Hi,1")
+    assertEquals(expected.sorted, StreamITCase.testResults.sorted)
+  }
+
+  @Test
+  def testEventTimeSlidingGroupWindowOverTime(): Unit = {
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)
+    val tEnv = TableEnvironment.getTableEnvironment(env)
+    StreamITCase.testResults = mutable.MutableList()
+
+    val data = new mutable.MutableList[(Long, Int, String)]
+    data.+=((1L, 1, "Hi"))
+    data.+=((2L, 2, "Hello"))
+    data.+=((4L, 2, "Hello"))
+    data.+=((8L, 3, "Hello world"))
+    data.+=((16L, 3, "Hello world"))
+    val stream = env
+      .fromCollection(data)
+      .assignTimestampsAndWatermarks(new TimestampWithEqualWatermark())
+
+    val table = stream.toTable(tEnv, 'long, 'int, 'string)
+
+    val windowedTable = table
+      .groupBy('string)
+      .window(Slide over 8.milli every 10.milli on 'rowtime)
```

Comment:
The DataStream doesn't limit that either. Maybe users want to do some sampling with a SlidingWindow. I wouldn't limit that.